init.lua

```diff
@@ -112,19 +114,39 @@ local KILL_OK = false
 local AREA_ACTIVATE = KILL_OK
 local AREA_NOACTIVATE = KILL_NO
 
+local savemodeToString = "FALSCH" -- only for debugging
 if safemode then
 	AREA_ACTIVATE = KILL_NO
 	AREA_NOACTIVATE = KILL_OK
+	savemodeToString = "WAHR"
 end
 
 -- Register punchplayer callback.
 minetest.register_on_punchplayer(function(player, hitter, time_from_last_punch, tool_capabilities, dir, damage)
+	local isPlayer = hitter:is_player()
+	
+	if hitter:is_player() == false then
+		return mobsDoNoDamage -- if this is a MOB then give Damage
+	end
+	--[[
+	local IsPlayerToString = "PLAYER"	-- only for debugging
+	if isplayer then 
+		IsPlayerToString = "PLAYER"			-- only for debugging
+	else
+		IsPlayerToString = "MOB"				-- only for debugging
+		return mobsDoNoDamage -- if this is a MOB then give Damage
+	end
+	]]--
+	
+	local playername = player:get_player_name() 
 	for k, v in pairs(pvp_areas_store:get_areas_for_pos(player:getpos())) do
 		if k then
-			return KILL_NO
+			--minetest.chat_send_player(playername, "in loop - safemode"..savemodeToString.." isPlayer "..IsPlayerToString)
+			return AREA_ACTIVATE --KILL_NO
```

Comment:
This is not the purpose of `AREA_ACTIVATE`. Use `KILL_NO` or `KILL_OK` when returning whether to damage.